error cannot find type definition file with typescript 4.5

![image](https://user-images.githubusercontent.com/45350572/142997233-f6c80c5d-764c-4ec9-b338-7f66f514fe16.png)

another related issue [vite-plugin-pages](https://github.com/hannoeru/vite-plugin-pages/issues/118)

Same issue for me, typescript 4.5.2 failed with build errors  Cannot find module '~icons/...' or its corresponding type declarations.

Had to revert to typescript 4.4.4 to resolve the issue

There is something wrong on the types, we need to review them.

@antfu see this, it seems we need to update the `package.json`, something  has changed (also applies to `vite-plugin-pages/client`):
```bash
======== Resolving type reference directive 'unplugin-icons/types/vue', containing file 'F:/work/projects/quini/GitHub/vitesse-with-ts-4.5/__inferred type names__.ts', root directory 'F:/work/projects/quini/GitHub/vitesse-with-ts-4.5/node_modules/@types'. ========
Resolving with primary search path 'F:/work/projects/quini/GitHub/vitesse-with-ts-4.5/node_modules/@types'.
Directory 'F:/work/projects/quini/GitHub/vitesse-with-ts-4.5/node_modules/@types/unplugin-icons/types' does not exist, skipping all lookups in it.
Looking up in 'node_modules' folder, initial location 'F:/work/projects/quini/GitHub/vitesse-with-ts-4.5'.
File 'F:/work/projects/quini/GitHub/vitesse-with-ts-4.5/node_modules/unplugin-icons/package.json' exists according to earlier cached lookups.
Export specifier './types/vue' does not exist in package.json scope at path 'F:/work/projects/quini/GitHub/vitesse-with-ts-4.5/node_modules/unplugin-icons'.
Directory 'F:/work/projects/quini/GitHub/node_modules' does not exist, skipping all lookups in it.
Directory 'F:/work/projects/quini/node_modules' does not exist, skipping all lookups in it.
Directory 'F:/work/projects/node_modules' does not exist, skipping all lookups in it.
Directory 'F:/work/node_modules' does not exist, skipping all lookups in it.
Directory 'F:/node_modules' does not exist, skipping all lookups in it.
======== Type reference directive 'unplugin-icons/types/vue' was not resolved. ========
```

Clone the repo here https://github.com/userquin/vitesse-with-ts-4.5 and run `pnpm install && tsc --traceResolution > ts-traceres.txt`, you will get this error (see it at hte bottom of the txt file). It will generate all `js` for each `ts` module:
```bash
error TS2688: Cannot find type definition file for 'unplugin-icons/types/vue'.
  The file is in the program because:
    Entry point of type library 'unplugin-icons/types/vue' specified in compilerOptions
error TS2688: Cannot find type definition file for 'vite-plugin-pages/client'.
  The file is in the program because:
    Entry point of type library 'vite-plugin-pages/client' specified in compilerOptions
```

Any updates on this? Have the same error but with **unplugin-icons/types/react**
```
error TS2688: Cannot find type definition file for 'unplugin-icons/types/react'.         
  The file is in the program because:                                                    
    Entry point of type library 'unplugin-icons/types/react' specified in compilerOptions
                                                                                         
  tsconfig.json:20:13                                                                    
    20             "unplugin-icons/types/react"                                          
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    File is entry point of type library specified here.
```

--- a/package.json
+++ b/package.json
@@ -37,6 +37,30 @@
       "require": "./dist/types.js",
       "import": "./dist/types.mjs"
     },
+    "./types/preact": {
+      "types": "./types/preact.d.ts"
+    },
+    "./types/raw": {
+      "types": "./types/raw.d.ts"
+    },
+    "./types/react": {
+      "types": "./types/react.d.ts"
+    },
+    "./types/solid": {
+      "types": "./types/solid.d.ts"
+    },
+    "./types/svelte": {
+      "types": "./types/svelte.d.ts"
+    },
+    "./types/vue": {
+      "types": "./types/vue.d.ts"
+    },
+    "./types/vue3": {
+      "types": "./types/vue3.d.ts"
+    },
+    "./types/web-components": {
+      "types": "./types/web-components.d.ts"
+    },
     "./vite": {
       "require": "./dist/vite.js",
       "import": "./dist/vite.mjs"

